### ISSUE TITLE
Enable MacOS and Windows testing on Travis-CI

### ISSUE THREAD

**Issue by DanHickstein** [PyAbel/PyAbel#323]
Travis now supports MacOS testing. Seems like a no-brainer that we should enable this and run at least one test on MacOS. Some significant fraction of our users (myself generally included) are on MacOS. 

Currently we are using Appveyor for Windows testing. This works fine, but it's another thing to maintain. We could keep the Windows testing here for now or try switching to Travis. I don't have strong feelings on this one.

**Comment by MikhailRyazanov** [comment]
I guess, adding MacOS testing to Travis definitely should be at least tried.

But I don't really understand their billing model. In [on place](https://docs.travis-ci.com/user/billing-faq/#what-if-i-am-building-open-source) they say:
> Each of the Travis CI Plans contains an amount of special OSS credits <u>per month</u> assigned to run builds only on public repositories.

But in [another](https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing#building-on-a-public-repositories-only):
> When your credit allotment runs out - we’d love for you to consider which of our plans will meet your needs.

and
> (should your run out of credits again you can repeat the process to request more or discuss a renewable amount)

So will you need to request renewals every time, or they allow us 25000 credits (=500 macOS minutes) _per month_?

**Comment by DanHickstein** [comment]
I think that smaller open-source projects just need to ask for one-off grants of credits. I don't think that our credits refresh every month. We currently have about 30k credits. Each build on travis takes about 90 seconds, so if we have 3 linux and 1 macOS build, then this is 10 * (4.5 mins) + 50 * (1.5 mins) = 120 credits. This is only 600 or so builds. So, we'll end up asking for more credits every year or two at our current rate. Probably if we pick up the pace we can ask for credits in larger chunks. 

Numpy and Scipy also got hit with the TravisCI paywall earlier this year:
https://github.com/scipy/scipy/issues/13109
https://github.com/numpy/numpy/issues/17824

Scipy seems to have dropped travis completely in favor of some GitHub built-in CI.

In our experience, Travis has been excellent, and I'd suggest that we keep using them unless the credit-requesting process becomes too onerous.

**Comment by MikhailRyazanov** [comment]
Sure, let's just keep it as is (with macOS enabled) for now and see how it goes. Maybe they'll change something again in the meantime... :–)

**Comment by MikhailRyazanov** [comment]
@DanHickstein, now (after #328) it seems that MacOS testing with the built-in Python 2.7 works, but we still don't know how to use Python 3 (and probably don't really need Travis-CI Windows testing). Would it be better to rename this issue or close it and open another, specific one, as a reminder?

**Comment by DanHickstein** [comment]
Yeah, I'll open a new issue for this.
